File: VB/AdaptiveSlideNavigation/UserControls/Navigation.ascx.vb
Imports System
Imports System.Web.UI
Imports System.Xml
Imports DevExpress.Web

Namespace AdaptiveSlideNavigation.UserControls
    Partial Public Class Navigation
        Inherits System.Web.UI.UserControl

        Protected Sub NavigationTreeView_NodeDataBound(ByVal source As Object, ByVal e As DevExpress.Web.TreeViewNodeEventArgs)
            Dim dataNode As XmlNode = (TryCast((TryCast(e.Node.DataItem, IHierarchyData)).Item, XmlNode))
            If dataNode.Name = "group" Then
                e.Node.NodeStyle.CssClass &= " group"
            End If
            If dataNode.ParentNode IsNot Nothing AndAlso dataNode.ParentNode.Name <> "group" Then
                e.Node.NodeStyle.CssClass &= " introPage"
            End If
        End Sub

        Protected Sub NavigationTreeView_PreRender(ByVal sender As Object, ByVal e As EventArgs)
            Dim node As TreeViewNode = NavigationTreeView.SelectedNode
            If node IsNot Nothing Then
                NavigationTreeView.ExpandToNode(node)
                node.Expanded = True
                Do While node IsNot Nothing
                    If node.Parent IsNot Nothing AndAlso node.Parent.Parent Is Nothing Then
                        node.Expanded = False
                    End If
                    node = node.Parent
                Loop
            End If
        End Sub
    End Class
End Namespace

File: CS/AdaptiveSlideNavigation/UserControls/Navigation.ascx.cs
using System;
using System.Web.UI;
using System.Xml;
using DevExpress.Web;

namespace AdaptiveSlideNavigation.UserControls {
    public partial class Navigation : System.Web.UI.UserControl {
        protected void NavigationTreeView_NodeDataBound(object source, DevExpress.Web.TreeViewNodeEventArgs e) {
            XmlNode dataNode = ((e.Node.DataItem as IHierarchyData).Item as XmlNode);
            if(dataNode.Name == "group")
                e.Node.NodeStyle.CssClass += " group";
            if(dataNode.ParentNode != null && dataNode.ParentNode.Name != "group")
                e.Node.NodeStyle.CssClass += " introPage";
        }

        protected void NavigationTreeView_PreRender(object sender, EventArgs e) {
            TreeViewNode node = NavigationTreeView.SelectedNode;
            if(node != null) {
                NavigationTreeView.ExpandToNode(node);
                node.Expanded = true;
                while(node != null) {
                    if(node.Parent != null && node.Parent.Parent == null)
                        node.Expanded = false;
                    node = node.Parent;
                }
            }
        }
    }
}